optima/results.py

```diff
@@ -487,57 +487,62 @@ def export(self, filename=None, folder=None, bypop=True, sep=',', ind=None, key=
-                if bypop and popkey!='tot': data = self.main[mainkey].pops[ind][pk-1,:] # WARNING, assumes 'tot' is always the first entry
+                if bypop and popkey != 'tot': data = self.main[mainkey].pops[ind][pk-1, :]  # WARNING, assumes 'tot' is always the first entry
                 else:                       data = self.main[mainkey].tot[ind][:]
                 outputstr += self.main[mainkey].name+sep+popkey+sep
                 for t in range(npts):
-                    if self.main[mainkey].ispercentage: outputstr += ('%s'+sep) % sigfig(data[t], sigfigs=sigfigs)
-                    else:                           outputstr += ('%i'+sep) % data[t]
-       
-        # Handle budget and coverage
-        thisbudget = []
-        thiscoverage = []
-        tvbudget = []
-        try:    thisbudget = self.budgets[key]
-        except: pass
-        try:    thiscoverage = self.coverages[key]
-        except: pass
-        try:    tvbudget = self.timevarying[key]
-        except: pass
-        
-        if len(thisbudget): # WARNING, does not support multiple years
-            outputstr += '\n\n\n'
-            outputstr += sep*2+'Budget\n'
-            outputstr += sep*2+sep.join(thisbudget.keys()) + '\n'
-            outputstr += sep*2+sep.join([str(val) for val in thisbudget.values()]) + '\n'
-        
-        if len(thiscoverage): # WARNING, does not support multiple years
-            covvals = thiscoverage.values()
-            for c in range(len(covvals)):
-                if checktype(covvals[c], 'arraylike'):
-                    covvals[c] = covvals[c][0] # Only pull out the first element if it's an array/list
-                if covvals[c] is None: covvals[c] = 0 # Just reset
-            outputstr += '\n\n\n'
-            outputstr += sep*2+'Coverage\n'
-            outputstr += sep*2+sep.join(thiscoverage.keys()) + '\n'
-            outputstr += sep*2+sep.join([str(val) for val in covvals]) + '\n' # WARNING, should have this val[0] but then dies with None entries
-
-        if len(tvbudget):
-            tvyears  = tvbudget['tvyears']
-            progkeys = tvbudget['tvbudgets'].keys()
-            tvdata   = tvbudget['tvbudgets'][:]
-            outputstr += '\n\n\n'
-            outputstr += sep*2+'Time-varying' + '\n'
-            outputstr += sep+sep.join(['Year']+progkeys) + '\n'
-            for y,year in enumerate(tvyears): # Loop over years as rows
-                outputstr += sep+str(year)+sep+sep.join([str(val) for val in tvdata[:,y]]) + '\n' # Join together programs as columns
+                    if '(per 100 p.y.)' in self.main[mainkey].name:
+                        outputstr += ('%s' + sep) % sigfig(100 * data[t], sigfigs=sigfigs)
+                    elif self.main[mainkey].ispercentage:
+                        outputstr += ('%sperc'+sep) % sigfig(data[t], sigfigs=sigfigs)
+                    else:
+                        outputstr += ('%i'+sep) % data[t]
+
+        if not comparisontab:
```

Comment:
If `comparisontab` is always true this will have to be rethought. Also, I would be concerned that this would break with trying to export results that don't have a budget (e.g. a Parscen).

Reply:
@cliffckerr: See above, we would want this code to be executed whenever it is an optimization multiresult